servers/physics_2d/godot_joints_2d.cpp

```diff
@@ -166,6 +167,9 @@ inline Vector2 custom_cross(const Vector2 &p_vec, real_t p_other) {
 }
 
 bool GodotPinJoint2D::pre_solve(real_t p_step) {
+	if (!are_constraints_enabled()) {
+		return true;
+	}
```

Comment:
These conditions are unnecessary as the constraints are removed from the bodies, and the physics step gets the constraints from the active bodies. The same goes for `solve`, in this joint and all the other joints, and also in 3D.